appcelerator/titanium_mobile
# fix(android): DateTimeFormat mishandles some combinations of settings on Android 8

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28073

**Note:**
We already have a unit test for this issue. That's how we discovered it.

**Test:**
1. Build and run the below on Android 8.
2. Verify the following error does **NOT** get logged on app startup.
`Failed to generate 'best' date pattern.`
3. Verify displayed date is: `03/31/2020, 8:15:30.123 PM`

```javascript
var date = new Date(Date.UTC(2020, 2, 31, 20, 15, 30, 123));
var formatter = new Intl.DateTimeFormat('en-US', {
	year: 'numeric',
	month: '2-digit',
	day: '2-digit',
	hour: 'numeric',
	minute: '2-digit',
	second: '2-digit',
	fractionalSecondDigits: 3,
	hour12: true,
	timeZone: 'UTC'
});
var window = Ti.UI.createWindow();
window.add(Ti.UI.createLabel({ text: formatter.format(date) }));
window.open();
```
